### ISSUE TITLE
Homebrew Reports Sourcery 2.0.0 is available, but there is no 2.0.0 release

### ISSUE THREAD

**Issue by rwgrier** [Homebrew/homebrew-core#122243]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
`brew config`:

brew config
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ddae5a6aced7a4178bdcc0d240d301227a548fc5
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode-14.2.0.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-x86_64
CLT: N/A
Xcode: 14.2 => /Applications/Xcode-14.2.0.app/Contents/Developer
```

`brew doctor`:
```
brew doctor
Your system is ready to brew.
```
```


### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I was attempting to update my formulae so that they are up to date. But I like to see what the changes are in the release notes for formulae like Sourcery. 

### What happened (include all command output)?

`brew update` said that there was a new version of [Sourcery](https://github.com/krzysztofzablocki/Sourcery) available (2.0.0). However, there is no 2.0.0 release on their [GitHub releases page](https://github.com/krzysztofzablocki/Sourcery/releases). 

 Could there have been a mistake in incrementing this formula?

`brew outdated`:
```
brew outdated
aom (3.5.0_1) < 3.5.0_2
apr (1.7.0_3) < 1.7.2
apr-util (1.6.1_4) < 1.6.3
cmake (3.25.1) < 3.25.2
fontconfig (2.14.1) < 2.14.2
freetds (1.3.16) < 1.3.17
highway (1.0.2) < 1.0.3
jpeg-turbo (2.1.4) < 2.1.5
jpeg-xl (0.7.0_1) < 0.8.0
sourcery (1.9.2) < 2.0.0
```

### What did you expect to happen?

I expected the formula to be marked as up to date since I have 1.9.2 installed which is the latest release. 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
`brew update`
`brew outdated`
```

**Comment by cho-m** [comment]
It shows up because the formula was updated to 2.0.0 in https://github.com/Homebrew/homebrew-core/pull/121068.

I'm guessing the git tag was removed, which means either 2.0.0 was a pre-release or it was yanked.

**Comment by iMichka** [comment]
I opened an issue upstream to get more insights what they plan is: https://github.com/krzysztofzablocki/Sourcery/issues/1139

**Comment by iMichka** [comment]
Will be fixed by #122329, please follow the discussion there.
